### ISSUE TITLE
RFE: "setAutoReduceSizeMode(boolean)" for Label

### ISSUE THREAD

**Issue by jsfan3** [codenameone/CodenameOne#2351]
At the moment, the Label class has the method "setAutoSizeMode(boolean)". It could be useful to have an option "setAutoReduceSizeMode(boolean)" that is very similar to "setAutoSizeMode(boolean)", except that it will not grow the text size, it will only reduce it if there isn't enough available space.

**Comment by codenameone** [comment]
This made sense to me when I accepted it but looking at the implementation it would be problematic. The problem is that we don't know the size to reduce from. E.g. say I have text of size X and I reduce it to fit. Then I rotate the device and have more space, which size do I return to. 

A simplistic solution might say to the size of the UIID but that would effectively block handcoding styles for that feature (which might be OK but it would make that code much harder). Instead I chose a simple yet effective solution to this problem: `setMaxAutoSize` which will allow you to define the maximum size for an autosize label in millimeters.
